src/com/spikes2212/genericsubsystems/drivetrains/commands/DriveArcadeWithPID.java

```diff
@@ -94,55 +105,79 @@ public DriveArcadeWithPID(TankDrivetrain drivetrain, PIDSource PIDSource, double
 	 */
 	public DriveArcadeWithPID(TankDrivetrain drivetrain, PIDSource PIDSource, Supplier<Double> setpointSupplier,
-			Supplier<Double> speedSupplier, PIDSettings PIDSettings) {
-		this(drivetrain, PIDSource, setpointSupplier, speedSupplier, () -> false, PIDSettings);
+			Supplier<Double> movementSupplier, PIDSettings PIDSettings, double outputRange) {
+		this(drivetrain, PIDSource, setpointSupplier, movementSupplier, () -> false, PIDSettings, outputRange);
 
 	}
 
 	/**
 	 * This constructs a new {@link DriveArcadeWithPID} ignoring the
 	 * {@link DriveArcadeWithPID#isFinishedSupplier} and uses constant values
 	 * for {@link DriveArcadeWithPID#setpointSupplier} and
-	 * {@link DriveArcadeWithPID#speedSupplier}
+	 * {@link DriveArcadeWithPID#movementSupplier}
 	 * 
 	 * @param drivetrain
 	 *            the {@link DriveArcadeWithPID} this command requires and moves
 	 * @param PIDSource
 	 *            the {@link PIDSource} that this command uses to get feedback
-	 *            about the {@link DriveArcadeWithPID#drivetrain}'s degree
+	 *            about the {@link DriveArcadeWithPID#drivetrain}'s position
 	 * @param setpoint
 	 *            constant value for {@link DriveArcadeWithPID#setpointSupplier}
-	 * @param speed
-	 *            constant value for {@link DriveArcadeWithPID#speedSupplier}
+	 * @param movement
+	 *            constant value for {@link DriveArcadeWithPID#movementSupplier}
 	 * @param PIDSettings
 	 *            {@link PIDSettings} for this command
+	 * @param outputRange
+	 *            the range of the source's output (for example, gyro's is 360)
 	 */
-	public DriveArcadeWithPID(TankDrivetrain drivetrain, PIDSource PIDSource, double setpoint, double speed,
-			PIDSettings PIDSettings) {
-		this(drivetrain, PIDSource, () -> setpoint, () -> speed, PIDSettings);
+	public DriveArcadeWithPID(TankDrivetrain drivetrain, PIDSource PIDSource, double setpoint, double movement,
+			PIDSettings PIDSettings, double outputRange) {
+		this(drivetrain, PIDSource, () -> setpoint, () -> movement, PIDSettings, outputRange);
 	}
 
 	@Override
 	protected void initialize() {
-		this.angleController = new PIDController(PIDSettings.getKP(), PIDSettings.getKI(), PIDSettings.getKD(), PIDSource,
-				(rotate) -> drivetrain.arcadeDrive(speedSupplier.get(), rotate));
-		angleController.setAbsoluteTolerance(PIDSettings.getTolerance());
-		angleController.setSetpoint(setpointSupplier.get());
-		angleController.setOutputRange(-1.0, 1.0);
-		angleController.enable();
+		this.rotationController = new PIDController(PIDSettings.getKP(), PIDSettings.getKI(), PIDSettings.getKD(),
+				PIDSource, (rotate) -> drivetrain.arcadeDrive(movementSupplier.get(), rotate / outputRange));
+		rotationController.setAbsoluteTolerance(PIDSettings.getTolerance());
+		rotationController.setSetpoint(setpointSupplier.get());
+		rotationController.setOutputRange(-outputRange / 2, outputRange / 2);
+		rotationController.enable();
```

Comment:
you still want the output range to be between -1 and 1 in order to move the motors, dont you?